### DISCUSSION TITLE
Does CS search through files like package-lock.json or yarn.lock?

### DISCUSSION THREAD

**Discussion by rathpc** [community/community#10361] [Other Feature Feedback, Questions, & Ideas] [answered]
Is there a list of files CS does not search through?

It seems like it never returns results within package-lock.json or yarn.lock files

**Comment by colinwm** [comment, marked as answer]
Hey @rathpc, we currently exclude files which we think are generated or vendored (including .lock files), see the FAQ here: https://cs.github.com/about/faq#indexed-content

**Reply by rathpc** [reply]
Hmm ok that's what I was afraid of. Maybe a feature could be added to include those optionally? It would be really helpful to check for versions of deps across an entire org codebases that may be susceptible to security concerns. This was at least my recent use case that I was unable to achieve due to the exclusion of those files.

**Reply by colinwm** [reply]
I've definitely heard of people searching yarn.lock files, e.g. for security. It's possible that we could change the rules, but we'll collect some more feedback first. Thanks for sharing!

**Reply by shivik** [reply]
Hello @colinwm, I am also looking for some feature by which we can include lock files

**Reply by donmckenna** [reply]
We'd also benefit from being able to search lock files, thanks @colinwm

**Reply by colinwm** [reply]
We're working on expanding the index to include all files, including lock files, though this is still in progress. Stay tuned!

**Reply by diegosempreboni** [reply]
Hi @colinwm, is there any update on this? We would really benefit to searching into yarn files.

**Reply by r0075h3ll** [reply]
+1 

@colinwm there are a lot of potential use-cases we can see getting implemented if `.lock` files are indexed.

**Comment by eyalroth** [comment]
@colinwm Please add an option to include all generated files (package lock or else).

I was not even aware that these are being excluded. This creates a "false sense of security".

Searching specifically in lock files is not just valuable for security reasons, but simply for identifying usage of packages/versions -- especially in an organization with many repositories.

**Comment by chris-grabcad** [comment]
Please 

1. Make it clear in the search and advanced search that these files are excluded, it's obviously tripping quite a few people up.
2. Either 
    1. Add an option to search to include generated files
    2. Update advanced search to include named files even if they're generated i.e. If I search for files called yarn.lock - then don't continue to exclude them!

With various worms and things becoming problematic Shai-Hulud for example - we should be able to search our repositories for references to compromised packages which is best done by searching through lock files.
